### ISSUE TITLE
[HOLD for payment 2022-12-20] [$250] BUG: Out of order LHN after split bill between web (staging.new.expensify) and IOS app reported by @chauchausoup

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11590]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Split bill with 7+ unique users on iOS/Android or web
2. Navigate back to the LHN
3. Observe that there is a different order of contacts between mobile app and Web
4. Observe that the group message has a different subtitle message between mobile app and Web

## Expected Result:
- Order of the contacts should be same in web and IOS of user profiles
- Both web and IOS should have correct subtitle messages.

## Actual Result:
Order of contacts is different in web and iOS

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android

**Version Number:** 1.2.11-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

<img width="1845" alt="image" src="https://user-images.githubusercontent.com/42940078/202967476-3dbe1218-1ef7-404a-895f-77cd45ae0a27.png">

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @chauchausoup
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1664646300696639

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sophiepintoraetz** [comment]
Hey @kavimuru - can you update the screenshots to clearly mark which chats you are referring to? It's hard to follow. Please reassign me once you've done so - or tag me in the comments.

**Comment by kavimuru** [comment]
@sophiepintoraetz I have attached the screenshot 
<img width="580" alt="splitbill order of contacts" src="https://user-images.githubusercontent.com/43996225/194120722-5634b5e6-53e6-40a5-9ade-48748886d13e.png">

**Comment by grgia** [comment]
I'm not able to replicate this on staging web/mobile

**Comment by sophiepintoraetz** [comment]
Hmm, actually, neither am I. I wonder if it's because it's happening on an older version (1.2.11-2 vs 1.2.11-3 - which is what I'm on).
<img width="835" alt="image" src="https://user-images.githubusercontent.com/42940078/194182418-be2e03a5-e9b8-495f-99eb-15b92c06e8d7.png">





Closing unless you can replicate it on 1.2.11-3.

**Comment by chauchausoup** [comment]
Yes I can reproduce it again at my end @sophiepintoraetz as per the video and image references attached above.

**Comment by sophiepintoraetz** [comment]
@chauchausoup - we need this to be reproducible, given you are the person who reported it, we need an independent reproduction.

**Comment by chauchausoup** [comment]
@sophiepintoraetz I think the maximum number of profiles that can take part in split bill is 8. So if you try your test with similar number of profiles then I think you might be able to reproduce.

**Comment by grgia** [comment]
@chauchausoup which version are you testing on?

**Comment by chauchausoup** [comment]
@grgia I am testing it with https://staging.new.expensify.com/ web app and IOS app from app store.

**Comment by sophiepintoraetz** [comment]
@grgia - this is in the OP `Version Number: 1.2.11-2` as well

**Comment by grgia** [comment]
Reopening, this is after requesting a split of $10 with 8 people. Different order and incorrect amount
<img width="1108" alt="Screen Shot 2022-10-17 at 10 39 52 AM" src="https://user-images.githubusercontent.com/38015950/196245717-17e22b0c-cf7e-4163-b031-c9051db7a0ba.png">
<img width="1480" alt="Screen Shot 2022-10-17 at 10 38 46 AM" src="https://user-images.githubusercontent.com/38015950/196245726-35db39b7-25b7-4870-bdc7-1523add4487c.png">
![IMG_4365](https://user-images.githubusercontent.com/38015950/196245786-48639eea-35a8-4a53-a6e6-647a1ee2ed8a.png)

**Comment by flodnv** [comment]
Maybe related...? https://github.com/Expensify/App/issues/11667

**Comment by mallenexpensify** [comment]
Putting on hold for a minute... https://github.com/Expensify/App/issues/11330 might fix this

**Comment by sophiepintoraetz** [comment]
Still on hold.

**Comment by sophiepintoraetz** [comment]
Still on Hold.

**Comment by sophiepintoraetz** [comment]
This issue isn't reproducible for me anymore - i think https://github.com/Expensify/App/issues/11330 fixed this. @eVoloshchak @chauchausoup or @kavimuru - can you?

**Comment by sophiepintoraetz** [comment]
All right - updated the OP because this only seems to happen with 7+ users in the split bill option - adding the help wanted label!
<img width="1845" alt="image" src="https://user-images.githubusercontent.com/42940078/202967020-9bb056ce-359b-414b-8322-30c148490fe1.png">

**Comment by sophiepintoraetz** [comment]
[Job posted!](https://www.upwork.com/jobs/~018ffc314ca48c00eb)

cc @rawalyogendra - given your work on https://github.com/Expensify/App/issues/11330?

**Comment by univalchemist** [comment]
@sophiepintoraetz  Can I have a chance to resolve this issue?
Two reasons might be possible for this issue.
- When split bill, data for contact might be changed
- For platforms, different ways to order might be applied.

**Comment by eVoloshchak** [comment]
@univalchemist, sure!
Could you please read [CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) and propose a solution for how you would resolve this issue?

**Comment by univalchemist** [comment]
First, I will get this app running on my local and reproduce this issue.
So
- I will need to run web and mobile
- You need to let me know what sort is correct in between web and app
- I will debug what happens when split bill with people
For now, I am not sure what is the exact reason for this issue technically but guess it is related to getting and sorting data from backend.

**Comment by eVoloshchak** [comment]
@univalchemist, please take a look at a couple of accepted proposals for the closed issues, [this one](https://github.com/Expensify/App/issues/11801#issuecomment-1278549911) for example. 

> You need to let me know what sort is correct in between web and app

That's a good question. As per [this](https://github.com/Expensify/App/blob/3528ad556579f3048eb91aa5ec02a2e98b35229b/src/libs/SidebarUtils.js#L112) comment:

```
// The LHN is split into five distinct groups, and each group is sorted a little differently. The groups will ALWAYS be in this order:
// 1. Pinned - Always sorted by reportDisplayName
// 2. Outstanding IOUs - Always sorted by iouReportAmount with the largest amounts at the top of the group
// 3. Drafts - Always sorted by reportDisplayName
// 4. Non-archived reports
//      - Sorted by lastMessageTimestamp in default (most recent) view mode
//      - Sorted by reportDisplayName in GSD (focus) view mode
// 5. Archived reports
//      - Sorted by lastMessageTimestamp in default (most recent) view mode
//      - Sorted by reportDisplayName in GSD (focus) view mode
```

**Comment by sophiepintoraetz** [comment]
@univalchemist - did you have a proposal you wanted to submit?

**Comment by univalchemist** [comment]
I need to run app on my local first and let me send you a proposal how to resolve. @sophiepintoraetz

**Comment by tjferriss** [comment]
Following instructions for the [weekly update chore](https://stackoverflow.com/c/expensify/questions/8208):

@flodnv this is one of the oldest issues in the /App repo. To help us clear out the large backlog of bugs, can you:

- Decide whether any proposals currently meet our guidelines and can be approved as-is
- For any that can't, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #bug-zero

**Comment by chiragsalian** [comment]
Investigating. Isn't the problem in this part,
> Outstanding IOUs - Always sorted by iouReportAmount with the largest amounts at the top of the group

I think if multiple IOUs have the same amount then there isn't a secondary sort to fallback on which is why we're getting different options across mobile and web. I'll try implementing a double sort and we can test it out.

**Comment by chiragsalian** [comment]
I've got fixes for the LHN ready. I've tested it and I think it works better.
I'd like to add some jest unit tests for it and then it should be ready for review. I'll add it on Monday since I'm OOO the rest of the week.

**Comment by chiragsalian** [comment]
Also another issue was mentioned here i.e., 
> Observe that the group message has a different subtitle message between mobile app and Web

Mind if we split that into another issue? You can assign it to me. I'd just like to tackle stuff in smaller parts 🙂

**Comment by eVoloshchak** [comment]
Not overdue, progress is being made in https://github.com/Expensify/App/pull/12978

**Comment by sophiepintoraetz** [comment]
@chiragsalian - did you have an idea on when you'll be able to get the PR out?

**Comment by chiragsalian** [comment]
So some higher priority deprecate sequence number tasks came up so I pushed this back a little bit. Still aiming for this week. 

Also @sophiepintoraetz can you split this part into another issue,
> Observe that the group message has a different subtitle message between mobile app and Web

**Comment by chiragsalian** [comment]
Updated the PR with tests and its now in review.

**Comment by chiragsalian** [comment]
Question, 
- so the problem right now is when two IOU amounts are the same the ordering of items on different devices can show up differently. 
- my solution for this is adding a logic where if IOU amounts are the same then sort by their names so that the ordering is improved on different devices.

So my question is, what if the report display names are also the same,
1. Should I add logic to handle this scenario, i.e., having a 3rd fallback sort like on `lastActionCreated` if display names are the same? (which feels a bit like a rabbit hole cause what if `lastActionCreated` is also the same) Or,
2.  Do nothing for now and add that if it's a problem in the future?

**Comment by chiragsalian** [comment]
Friendly bump @SofiedeVreese or @tjferriss. Any thoughts on my question above?

**Comment by tjferriss** [comment]
>Do nothing for now and add that if it's a problem in the future?

I agree with doing nothing here as this seems like a rare possibility.

**Comment by SofiedeVreese** [comment]
cc @sophiepintoraetz , not me 🤣

**Comment by chiragsalian** [comment]
oh whoops 😅

**Comment by sophiepintoraetz** [comment]
Got it! invited you to the job.
